File: scripts/smoke-test.mjs
/**
 * Pre-deploy check: can the app reach monday.com and Gemini?
 *
 *   node --env-file=.env.local scripts/smoke-test.mjs
 *   node --env-file=.env.local scripts/smoke-test.mjs --all
 */

const { GEMINI_MODEL, SYSTEM_PROMPT } = await import("../lib/config.ts");
const { DEALS_BOARD_ID, WORK_ORDERS_BOARD_ID, MONDAY_API_URL } = await import("../lib/config.ts");

const QUESTIONS = [
  "List the live monday boards you received and how many rows each snapshot contains. Do not analyze.",
  "How's our deal pipeline looking by sector? Flag any data quality issues you hit.",
  "What's our billing and collection status across work orders? Anything at risk?",
  "Which sectors have deals but no active work orders?",
  "Prepare a leadership update on pipeline health.",
];

const missing = ["GEMINI_API_KEY", "MONDAY_TOKEN"].filter((key) => !process.env[key]);
if (missing.length) {
  console.error(`Missing env vars: ${missing.join(", ")}`);
  process.exit(1);
}

const BOARD_QUERY = `
  query BoardSnapshot($ids: [ID!]!) {
    boards(ids: $ids) {
      id
      name
      items_count
      columns { id title type }
      items_page(limit: 500) {
        items {
          id
          name
          column_values { id text type }
        }
      }
    }
  }
`;

async function fetchSnapshot() {
  const response = await fetch(MONDAY_API_URL, {
    method: "POST",
    headers: {
      Authorization: process.env.MONDAY_TOKEN,
      "API-Version": "2026-07",
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      query: BOARD_QUERY,
      variables: { ids: [String(DEALS_BOARD_ID), String(WORK_ORDERS_BOARD_ID)] },
    }),
  });

  const json = await response.json();
  if (!response.ok || json.errors?.length) {
    throw new Error(json.errors?.map((error) => error.message).join("; ") || `monday status ${response.status}`);
  }

  const normalize = (board) => {
    const titleById = new Map(board.columns.map((column) => [column.id, column.title]));
    return {
      id: Number(board.id),
      liveName: board.name,
      itemCount: board.items_count,
      fetchedCount: board.items_page.items.length,
      rows: board.items_page.items.map((item) => {
        const values = {};
        for (const value of item.column_values) {
          const text = value.text?.trim();
          if (text) values[titleById.get(value.id) || value.id] = text;
        }
        return { id: item.id, name: item.name, values };
      }),
    };
  };

  const boardsById = new Map(json.data.boards.map((board) => [Number(board.id), board]));
  return {
    fetchedAt: new Date().toISOString(),
    boards: {
      deals: normalize(boardsById.get(DEALS_BOARD_ID)),
      workOrders: normalize(boardsById.get(WORK_ORDERS_BOARD_ID)),
    },
  };
}

const snapshot = await fetchSnapshot();
const snapshotJson = JSON.stringify(snapshot);

async function ask(question) {
  const response = await fetch(
    `https://generativelanguage.googleapis.com/v1beta/models/${GEMINI_MODEL}:generateContent`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "x-goog-api-key": process.env.GEMINI_API_KEY,
      },
      body: JSON.stringify({
        systemInstruction: { parts: [{ text: SYSTEM_PROMPT }] },
        contents: [
          {
            role: "user",
            parts: [{ text: `LIVE MONDAY SNAPSHOT JSON:\n${snapshotJson}\n\nUSER: ${question}` }],
          },
        ],
        generationConfig: {
          temperature: 0.2,
          maxOutputTokens: 4096,
        },
      }),
    },
  );

  const json = await response.json();
  if (!response.ok || json.error?.message) {
    throw new Error(json.error?.message || `Gemini status ${response.status}`);
  }

  return json.candidates?.[0]?.content?.parts?.map((part) => part.text || "").join("").trim();
}

const all = process.argv.includes("--all");
const suite = all ? QUESTIONS : QUESTIONS.slice(0, 1);
let clean = true;

console.log(
  `Read ${snapshot.boards.deals.fetchedCount}/${snapshot.boards.deals.itemCount} deal rows and ` +
    `${snapshot.boards.workOrders.fetchedCount}/${snapshot.boards.workOrders.itemCount} work-order rows.`,
);

for (const [i, question] of suite.entries()) {
  console.log(`\n${"=".repeat(70)}\nQ${i + 1}: ${question}\n${"-".repeat(70)}`);
  try {
    console.log(await ask(question));
  } catch (err) {
    console.error(`\n!! Failed: ${err.message}`);
    clean = false;
  }
}

console.log(`\n${"=".repeat(70)}`);
if (!clean) {
  console.error("Smoke test FAILED - fix before deploying.");
  process.exit(1);
}
console.log(all ? "All questions answered from live board data." : "Connection OK. Run with --all for the full suite.");

File: smoke_test.py
"""Pre-deploy check: can the agent actually reach both monday.com boards?

Run this the moment you have MONDAY_TOKEN + ANTHROPIC_API_KEY, before building
anything else on top. MCP auth failures surface only when a tool call runs, so
a broken token looks fine until mid-demo.

    python smoke_test.py
"""
import os
import sys

import anthropic

from config import MODEL, MONDAY_MCP_URL, SYSTEM_PROMPT

# Doubles as the manual test script for the hosted app — run these in the
# deployed UI too. Each targets one PRD user story.
QUESTIONS = [
    "List the boards you can see and the row count of each. Don't analyze, just confirm access.",
    "How's our deal pipeline looking by sector? Flag any data quality issues you hit.",
    "What's our billing and collection status across work orders? Anything at risk?",
    "Which sectors have deals but no active work orders?",
    "Prepare a leadership update on pipeline health.",
]


def ask(client, question):
    resp = client.beta.messages.create(
        model=MODEL,
        max_tokens=4096,
        betas=["mcp-client-2025-11-20"],
        system=SYSTEM_PROMPT,
        mcp_servers=[{
            "type": "url",
            "url": MONDAY_MCP_URL,
            "name": "monday",
            "authorization_token": os.environ["MONDAY_TOKEN"],
        }],
        tools=[{"type": "mcp_toolset", "mcp_server_name": "monday"}],
        messages=[{"role": "user", "content": question}],
    )
    text = "".join(b.text for b in resp.content if b.type == "text")
    used_mcp = any(b.type in ("mcp_tool_use", "mcp_tool_result") for b in resp.content)
    return text, used_mcp


def main():
    missing = [k for k in ("ANTHROPIC_API_KEY", "MONDAY_TOKEN") if k not in os.environ]
    if missing:
        sys.exit(f"Missing env vars: {', '.join(missing)}")

    client = anthropic.Anthropic()
    only_first = "--all" not in sys.argv

    for i, q in enumerate(QUESTIONS[:1] if only_first else QUESTIONS, 1):
        print(f"\n{'=' * 70}\nQ{i}: {q}\n{'-' * 70}")
        text, used_mcp = ask(client, q)
        print(text)
        if not used_mcp:
            print("\n!! Answered without calling monday.com — check the MCP connection.")

    if only_first:
        print(f"\n{'=' * 70}\nConnection OK. Run with --all for the full 5-question suite.")


if __name__ == "__main__":
    main()
